src/viam/rpc/server.py

```diff
@@ -45,6 +46,24 @@ def __init__(self, components: List[ResourceBase], *, module_service: Optional["
         if module_service is not None:
             services.append(module_service)
         services = ServerReflection.extend(services)
+
+        # Go into every service and replace the methods with a wrapped method that has error handling
+        for service in services:
+
+            def update_mapping():
```

Comment:
[q] Why did you put create a function to do this instead of just keeping it in the body of the loop?

Reply:
This is done so that the updating maps happens during runtime. Doing it beforehand in the loop caused it so that `mapping` pushed into the server was the last `mapping` value.

Reply:
Got it! That makes sense.

Reply:
Thank you!